classes/task/report_process.php

```diff
@@ -492,11 +516,25 @@ private function calculate_total_conversion_cost(
 
         $convertfrom = time() - (int)get_config('local_smartmedia', 'convertfrom');
 
-        // Get the location pricing for the AWS region set.
-        $location = get_config('local_smartmedia', 'api_region');
-        $transcodelocationpricing = $transcodepricingclient->get_location_pricing($location);
-        $rekoglocationpricing = $rekogpricingclient->get_location_pricing($location);
-        $transcribelocationpricing = $transcribepricingclient->get_location_pricing($location);
+        // Check if we have already cached the pricing.
+        if (empty($this->pricing)) {
+            // Get the location pricing for the AWS region set.
+            $location = get_config('local_smartmedia', 'api_region');
+            $transcodelocationpricing = $transcodepricingclient->get_location_pricing($location);
+            $rekoglocationpricing = $rekogpricingclient->get_location_pricing($location);
+            $transcribelocationpricing = $transcribepricingclient->get_location_pricing($location);
+
+            $this->pricing = [
+                'transcode' => $transcodelocationpricing,
+                'rekog' => $rekoglocationpricing,
+                'transcribe' => $transcribelocationpricing
+            ];
+        } else {
+            $transcodelocationpricing = $this->pricing['transcode'];
+            $rekoglocationpricing = $this->pricing['rekog'];
+            $transcribelocationpricing = $this->pricing['transcribe'];
+        }
+
```

Comment:
Hmm, I feel like it's a bit hard to track where values might be coming from if skimming. I suggest it might be clearer to handle caching by itself, then always derive the values from that cache.

```php
// Check if we have already cached the pricing.
if (empty($this->pricing)) {
    // Get the location pricing for the AWS region set.
    $location = get_config('local_smartmedia', 'api_region');
    $this->pricing = [
        'transcode' => $transcodepricingclient->get_location_pricing($location),
        'rekog' => $rekogpricingclient->get_location_pricing($location),
        'transcribe' => $transcribepricingclient->get_location_pricing($location)
    ];
}
    
$transcodelocationpricing = $this->pricing['transcode'];
$rekoglocationpricing = $this->pricing['rekog'];
$transcribelocationpricing = $this->pricing['transcribe'];

```

What do you think?